packager/packager.cc

```diff
@@ -725,46 +795,122 @@ class ImportTree final {
 class ImportTreeBuilder final {
     // PackageInfoImpl package; // The package we are building an import tree for.
     core::NameRef pkgMangledName;
+    core::NameRef privatePkgMangledName;
     ImportTree root;
 
 public:
-    ImportTreeBuilder(const PackageInfoImpl &package) : pkgMangledName(package.name.mangledName) {}
+    ImportTreeBuilder(const PackageInfoImpl &package)
+        : pkgMangledName(package.name.mangledName), privatePkgMangledName(package.privateMangledName) {}
     ImportTreeBuilder(const ImportTreeBuilder &) = delete;
     ImportTreeBuilder(ImportTreeBuilder &&) = default;
     ImportTreeBuilder &operator=(const ImportTreeBuilder &) = delete;
     ImportTreeBuilder &operator=(ImportTreeBuilder &&) = default;
 
-    void mergeImports(const PackageInfoImpl &importedPackage, const Import &import) {
-        for (const auto &exp : importedPackage.exports) {
-            if (exp.type == ExportType::Public) {
-                addImport(importedPackage, import.name.loc, exp.fqn, import.type);
+    void mergeImports(core::Context ctx, const PackageInfoImpl &package) {
```

Comment:
We really could use some method-level comments on some of these to indicate what they're doing. I don't know that it's necessary to add them as part of this PR, but e.g. a method named something like `mergeImports` tells me very little about what it does and how.

Reply:
Sure, can add some more comments.